app/domain/qualification/expiration.rb

```diff
@@ -0,0 +1,33 @@
+# frozen_string_literal: true
+
+#  Copyright (c) 2024, Schweizer Alpen-Club. This file is part of
+#  hitobito_sac_cas and licensed under the Affero General Public License version 3
+#  or later. See the COPYING file at the top-level directory or at
+#  https://github.com/hitobito/hitobito_sac_cas
+
+class Qualification::Expiration
+  class << self
+    def send_reminder(time, finish_at)
+      content_key = Qualification::ExpirationMailer.const_get :"REMINDER_#{time}"
+
+      Qualification.where(finish_at: finish_at)
+                   .joins(latest_expirations_join)
+                   .includes(:person)
+                   .each do |qualification|
+        Qualification::ExpirationMailer.reminder(qualification.person, content_key).deliver_now
+      end
+    end
+
+    private
+
+    def latest_expirations
+      Qualification.select('MAX(finish_at) as finish_at, person_id').group(:person_id).to_sql
+    end
+
+    def latest_expirations_join
+      "INNER JOIN (#{latest_expirations}) as l" \
```

Comment:
```suggestion
      "INNER JOIN (#{latest_expiring_qualifications}) as latest_expiring_qualifications" \
```

ich würde das hier ganz auschreiben damit es verständlich bleibt